packages/framework/presence/src/test/presenceManager.spec.ts

```diff
@@ -100,6 +100,46 @@ describe("Presence", () => {
 				let initialAttendeeSignal: ReturnType<typeof generateBasicClientJoin>;
 				let rejoinAttendeeSignal: ReturnType<typeof generateBasicClientJoin>;
 
+				// Processes join signals and returns the attendees that were announced via `attendeeJoined`
+				function processJoinSignals(
+					signals: ReturnType<typeof generateBasicClientJoin>[],
+				): ISessionClient[] {
+					const joinedAttendees: ISessionClient[] = [];
+					const cleanUpListener = presence.events.on("attendeeJoined", (attendee) => {
+						joinedAttendees.push(attendee);
+					});
+
+					for (const signal of signals) {
+						presence.processSignal("", signal, false);
+					}
+
+					cleanUpListener();
+					return joinedAttendees;
+				}
+
+				function verifyAttendee(
+					actualAttendee: ISessionClient,
+					expectedConnectionId: ClientConnectionId,
+					expectedSessionId: string,
+					expectedConnectionStatus: SessionClientStatus = SessionClientStatus.Connected,
+				): void {
+					assert.equal(
+						actualAttendee.sessionId,
+						expectedSessionId,
+						"Attendee has wrong session id",
+					);
+					assert.equal(
+						actualAttendee.getConnectionId(),
+						expectedConnectionId,
+						"Attendee has wrong client connection id",
+					);
+					assert.equal(
+						actualAttendee.getConnectionStatus(),
+						expectedConnectionStatus,
+						"Attendee connection status is not Connected",
```

Comment:
expected status might not be Connected. You need to say "is not ${expectedConnectionStatus}".